src/librustc/session/mod.rs

```diff
@@ -776,7 +776,10 @@ pub fn build_session_(sopts: config::Options,
     let print_fuel_crate = sopts.debugging_opts.print_fuel.clone();
     let print_fuel = Cell::new(0);
 
-    let working_dir = env::current_dir().unwrap().to_string_lossy().into_owned();
+    let working_dir = match env::current_dir() {
+        Ok(dir) => dir.to_string_lossy().into_owned(),
+        Err(e) => panic!(p_s.span_diagnostic.fatal(&format!("Current directory is invalid: {}", e))),
```

Comment:
This line is currently too long. This repository requires a 100-line limit.

```
[00:03:17] tidy error: /checkout/src/librustc/session/mod.rs:781: line longer than 100 chars
```

Reply:
Oops fixed in https://github.com/rust-lang/rust/pull/45505/commits/cfc916ebf8a9039e321924ed111727c2fcfb3e70